MailContainerTest.Tests/MailTransferServiceTests.cs

```diff
@@ -0,0 +1,61 @@
+using MailContainerTest.Data;
+using MailContainerTest.Services;
+using MailContainerTest.Types;
+using Moq;
+using NUnit.Framework;
+
+namespace MailContainerTest.Tests
+{
+    public class MailTransferServiceTests
+    {
+        [Test]
+        public void MakeMailTransfer_TransferAllowed_ContainersAreUpdated()
+        {
+            var dataStore = new Mock<IMailContainerDataStore>();
+
+            dataStore
+                .Setup(m => m.GetMailContainer(It.IsAny<string>()))
+                .Returns(new MailContainer());
+
+            var requestApprovalService = new Mock<IMakeMailTransferRequestApprovalService>();
+
+            requestApprovalService
+                .Setup(m => m.TransferIsAllowed(It.IsAny<MailContainer>(), It.IsAny<MakeMailTransferRequest>()))
+                .Returns(true);
+
+            var transferService = new MailTransferService(dataStore.Object, requestApprovalService.Object);
+
+            var result = transferService.MakeMailTransfer(new MakeMailTransferRequest());
+
+            dataStore.Verify(
+                m => m.UpdateMailContainer(It.IsAny<MailContainer>()),
+                Times.Exactly(2)
+            );
```

Comment:
Here I would also verify that the capacities of the source and destination containers had been updated appropriately.